kornia/enhance/equalization.py

```diff
@@ -0,0 +1,306 @@
+    t = tiw * (tl - tr) + tr
+    b = tiw * (bl - br) + br
+    tiles_equalized[:, 1:-1, 1:-1] = tih * (t - b) + b
+
+    # corner regions
+    tiles_equalized[:, 0::gh - 1, 0::gw - 1] = preinterp_tiles_equalized[:, 0::gh - 1, 0::gw - 1, 0]
+
+    # border region (h)
+    t, b, _, _ = preinterp_tiles_equalized[:, 1:-1, 0].unbind(2)
+    tiles_equalized[:, 1:-1, 0] = tih.squeeze(1) * (t - b) + b
+    t, b, _, _ = preinterp_tiles_equalized[:, 1:-1, gh - 1].unbind(2)
+    tiles_equalized[:, 1:-1, gh - 1] = tih.squeeze(1) * (t - b) + b
+
+    # border region (w)
+    l, r, _, _ = preinterp_tiles_equalized[:, 0, 1:-1].unbind(2)
+    tiles_equalized[:, 0, 1:-1] = tiw * (l - r) + r
+    l, r, _, _ = preinterp_tiles_equalized[:, gw - 1, 1:-1].unbind(2)
+    tiles_equalized[:, gw - 1, 1:-1] = tiw * (l - r) + r
+
+    # same type as the input
+    return tiles_equalized.to(interp_tiles).div(255.)
+
+
+def equalize_clahe(input: torch.Tensor, clip_limit: float = 40., grid_size: Tuple[int, int] = (8, 8)) -> torch.Tensor:
+    r"""Apply clahe equalization on the input tensor.
+
+    Args:
+        input (torch.Tensor): images tensor to equalize with shapes like :math:`(C, H, W)` or :math:`(B, C, H, W)`.
+        clip_limit (float): threshold value for contrast limiting. If 0 clipping is disabled. Default: 40.
+        grid_size (Tuple[int, int]): number of tiles to be cropped in each direction (GH, GW).
+
+    Returns:
+        torch.Tensor: Equalized image or images with shape as the input.
+
+    """
+    if not torch.is_tensor(input):
+        raise TypeError(f"Input input type is not a torch.Tensor. Got {type(input)}")
+    if input.dim() not in [3, 4]:
+        raise ValueError(f"Invalid input shape, we expect CxHxW or BxCxHxW. Got: {input.shape}")
+    if input.numel() == 0:
+        raise ValueError(f"Invalid input tensor, it is empty.")
+    if not isinstance(clip_limit, float):
+        raise TypeError(f"Input clip_limit type is not float. Got {type(clip_limit)}")
+    if not isinstance(grid_size, tuple):
+        raise TypeError(f"Input grid_size type is not Tuple. Got {type(grid_size)}")
+    if len(grid_size) != 2:
+        raise TypeError(f"Input grid_size is not a Tuple with 2 elements. Got {len(grid_size)}")
+    if type(grid_size[0]) != type(grid_size[1]) or isinstance(grid_size[0], float):
+        raise TypeError("Input grid_size type is not valid, must be a Tuple[int, int].")
+    if grid_size[0] <= 0 or grid_size[1] <= 0:
+        raise ValueError("Input grid_size elements must be positive. Got {grid_size}")
+
+    imgs: torch.Tensor = _to_bchw(input)  # B x C x H x W
+
+    hist_tiles: torch.Tensor  # B x GH x GW x C x TH x TW
+    img_padded: torch.Tensor  # B x C x H' x W'
+    # the size of the tiles must be even in order to divide them into 4 tiles for the interpolation
+    hist_tiles, img_padded = _compute_tiles(imgs, grid_size, True)
+    tile_size: Tuple[int, int] = hist_tiles.shape[-2:]  # type: ignore
```

Comment:
why ignore?

Reply:
mypy cannot infer the type. I added a comment.